File: app/src/main/java/com/alaskalany/careershowcase/database/DataGenerator.java
/*
 * MIT License
 *
 * Copyright (c) 2018 Ahmed AlAskalany
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.alaskalany.careershowcase.database;

import androidx.annotation.NonNull;
import com.alaskalany.careershowcase.entity.ContactEntity;
import com.alaskalany.careershowcase.entity.EducationEntity;
import com.alaskalany.careershowcase.entity.SkillEntity;
import com.alaskalany.careershowcase.entity.WorkEntity;
import org.jetbrains.annotations.Contract;

import java.util.*;

public class DataGenerator {
    
    @NonNull
    @Contract(pure = true)
    public static List<ContactEntity> generateContacts() {
        
        return ContactContent.ITEMS;
    }
    
    @NonNull
    @Contract(pure = true)
    public static List<EducationEntity> generateEducations() {
        
        return EducationContent.ITEMS;
    }
    
    @NonNull
    @Contract(pure = true)
    public static List<SkillEntity> generateSkills() {
        
        return SkillContent.ITEMS;
    }
    
    @NonNull
    @Contract(pure = true)
    public static List<WorkEntity> generateWorks() {
        
        return WorkContent.ITEMS;
    }
    
    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     * <p>
     * TODO: Replace all uses of this class before publishing your app.
     */
    public static class ContactContent {
        
        /**
         * An array of sample (contact) items.
         */
        public static final List<ContactEntity> ITEMS = new ArrayList<>();
        
        /**
         * A map of sample (contact) items, by ID.
         */
        @SuppressWarnings("WeakerAccess")
        public static final Map<Integer, ContactEntity> ITEM_MAP = new HashMap<Integer, ContactEntity>();
        
        private static final int COUNT = 25;
        
        static {
            // Add some sample items.
            for (int i = 1; i <= COUNT; i++) {
                addItem(createContactItem(i));
            }
        }
        
        private static void addItem(ContactEntity item) {
            
            ITEMS.add(item);
            ITEM_MAP.put(item.getId(), item);
        }
        
        @NonNull
        private static ContactEntity createContactItem(int position) {
            
            return new ContactEntity(position, "Item " + position, makeDetails(position));
        }
        
        @NonNull
        private static String makeDetails(int position) {
            
            StringBuilder builder = new StringBuilder();
            builder.append("Details about Item: ")
                   .append(position);
            for (int i = 0; i < position; i++) {
                builder.append("\nMore contactDetails information here.");
            }
            return builder.toString();
        }
    }
    
    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     * <p>
     * TODO: Replace all uses of this class before publishing your app.
     */
    public static class EducationContent {
        
        /**
         * An array of sample (education) items.
         */
        public static final List<EducationEntity> ITEMS = new ArrayList<>();
        
        private static final int COUNT = 25;
        
        static {
            // Add some sample items.
            for (int i = 1; i <= COUNT; i++) {
                addItem(createEducationItem(i));
            }
        }
        
        private static void addItem(EducationEntity item) {
            
            ITEMS.add(item);
        }
        
        @NonNull
        private static EducationEntity createEducationItem(int position) {
            
            EducationEntity educationEntity = new EducationEntity();
            educationEntity.setId(position);
            educationEntity.setTitle("Education " + position);
            educationEntity.setDescription(makeDetails(position));
            educationEntity.setInstitution("KTH Royal Institute of Technology");
            educationEntity.setLocation("Stockholm, Sweden");
            educationEntity.setDuration("2014-2019");
            return educationEntity;
        }
        
        @NonNull
        private static String makeDetails(int position) {
            
            StringBuilder builder = new StringBuilder();
            builder.append("Details about Education: ")
                   .append(position);
            for (int i = 0; i < position; i++) {
                builder.append("\nMore contactDetails information here.");
            }
            return builder.toString();
        }
    }
    
    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     * <p>
     * TODO: Replace all uses of this class before publishing your app.
     */
    public static class SkillContent {
        
        /**
         * An array of sample (skill) items.
         */
        public static final List<SkillEntity> ITEMS = new ArrayList<>();
        
        private static final int COUNT = 25;
        
        static {
            // Add some sample items.
            for (int i = 1; i <= COUNT; i++) {
                addItem(createSkillItem(i));
            }
        }
        
        private static void addItem(SkillEntity item) {
            
            ITEMS.add(item);
        }
        
        @NonNull
        private static SkillEntity createSkillItem(int position) {
            
            SkillEntity skillEntity = new SkillEntity();
            skillEntity.setId(position);
            skillEntity.setTitle("Skill " + position);
            skillEntity.setLevel(makeLevel());
            return skillEntity;
        }
        
        private static int makeLevel() {
            
            return new Random().nextInt(5);
        }
    }
    
    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     * <p>
     * TODO: Replace all uses of this class before publishing your app.
     */
    public static class WorkContent {
        
        /**
         * An array of sample (contact) items.
         */
        public static final List<WorkEntity> ITEMS = new ArrayList<>();
        
        private static final int COUNT = 25;
        
        static {
            // Add some sample items.
            for (int i = 1; i <= COUNT; i++) {
                addItem(createWorkItem(i));
            }
        }
        
        private static void addItem(WorkEntity item) {
            
            ITEMS.add(item);
        }
        
        @NonNull
        private static WorkEntity createWorkItem(int position) {
            
            WorkEntity workEntity = new WorkEntity();
            workEntity.setId(position);
            workEntity.setTitle("Work " + position);
            workEntity.setDescription(makeDetails(position));
            return workEntity;
        }
        
        @NonNull
        private static String makeDetails(int position) {
            
            StringBuilder builder = new StringBuilder();
            builder.append("Details about Work: ")
                   .append(position);
            for (int i = 0; i < position; i++) {
                builder.append("\nMore contactDetails information here.");
            }
            return builder.toString();
        }
    }
}

File: app/src/main/java/com/alaskalany/careershowcase/database/DataGenerator.kt
/*
 * MIT License
 *
 * Copyright (c) 2018 Ahmed AlAskalany
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.alaskalany.careershowcase.database

import com.alaskalany.careershowcase.entity.ContactEntity
import com.alaskalany.careershowcase.entity.EducationEntity
import com.alaskalany.careershowcase.entity.SkillEntity
import com.alaskalany.careershowcase.entity.WorkEntity
import org.jetbrains.annotations.Contract

import java.util.*

object DataGenerator {

    @JvmStatic
    @Contract(pure = true)
    fun generateContacts(): List<ContactEntity> {

        return ContactContent.ITEMS
    }

    @JvmStatic
    @Contract(pure = true)
    fun generateEducations(): List<EducationEntity> {

        return EducationContent.ITEMS
    }

    @JvmStatic
    @Contract(pure = true)
    fun generateSkills(): List<SkillEntity> {

        return SkillContent.ITEMS
    }

    @JvmStatic
    @Contract(pure = true)
    fun generateWorks(): List<WorkEntity> {

        return WorkContent.ITEMS
    }

    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     *
     *
     * TODO: Replace all uses of this class before publishing your app.
     */
    object ContactContent {

        /**
         * An array of sample (contact) items.
         */
        val ITEMS: MutableList<ContactEntity> = ArrayList()

        /**
         * A map of sample (contact) items, by ID.
         */
        val ITEM_MAP: MutableMap<Int, ContactEntity> = HashMap()

        private val COUNT = 25

        init {
            // Add some sample items.
            for (i in 1..COUNT) {
                addItem(createContactItem(i))
            }
        }

        private fun addItem(item: ContactEntity) {

            ITEMS.add(item)
            ITEM_MAP[item.id] = item
        }

        private fun createContactItem(position: Int): ContactEntity {

            return ContactEntity(position, "Item $position", makeDetails(position))
        }

        private fun makeDetails(position: Int): String {

            val builder = StringBuilder()
            builder.append("Details about Item: ")
                    .append(position)
            for (i in 0 until position) {
                builder.append("\nMore contactDetails information here.")
            }
            return builder.toString()
        }
    }

    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     *
     *
     * TODO: Replace all uses of this class before publishing your app.
     */
    object EducationContent {

        /**
         * An array of sample (education) items.
         */
        val ITEMS: MutableList<EducationEntity> = ArrayList()

        private val COUNT = 25

        init {
            // Add some sample items.
            for (i in 1..COUNT) {
                addItem(createEducationItem(i))
            }
        }

        private fun addItem(item: EducationEntity) {

            ITEMS.add(item)
        }

        private fun createEducationItem(position: Int): EducationEntity {

            val educationEntity = EducationEntity()
            educationEntity.id = position
            educationEntity.title = "Education $position"
            educationEntity.description = makeDetails(position)
            educationEntity.institution = "KTH Royal Institute of Technology"
            educationEntity.location = "Stockholm, Sweden"
            educationEntity.duration = "2014-2019"
            return educationEntity
        }

        private fun makeDetails(position: Int): String {

            val builder = StringBuilder()
            builder.append("Details about Education: ")
                    .append(position)
            for (i in 0 until position) {
                builder.append("\nMore contactDetails information here.")
            }
            return builder.toString()
        }
    }

    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     *
     *
     * TODO: Replace all uses of this class before publishing your app.
     */
    object SkillContent {

        /**
         * An array of sample (skill) items.
         */
        val ITEMS: MutableList<SkillEntity> = ArrayList()

        private val COUNT = 25

        init {
            // Add some sample items.
            for (i in 1..COUNT) {
                addItem(createSkillItem(i))
            }
        }

        private fun addItem(item: SkillEntity) {

            ITEMS.add(item)
        }

        private fun createSkillItem(position: Int): SkillEntity {

            val skillEntity = SkillEntity()
            skillEntity.id = position
            skillEntity.title = "Skill $position"
            skillEntity.level = makeLevel()
            return skillEntity
        }

        private fun makeLevel(): Int {

            return Random().nextInt(5)
        }
    }

    /**
     * Helper class for providing sample description for user interfaces created by
     * Android template wizards.
     *
     *
     * TODO: Replace all uses of this class before publishing your app.
     */
    object WorkContent {

        /**
         * An array of sample (contact) items.
         */
        val ITEMS: MutableList<WorkEntity> = ArrayList()

        private val COUNT = 25

        init {
            // Add some sample items.
            for (i in 1..COUNT) {
                addItem(createWorkItem(i))
            }
        }

        private fun addItem(item: WorkEntity) {

            ITEMS.add(item)
        }

        private fun createWorkItem(position: Int): WorkEntity {

            val workEntity = WorkEntity()
            workEntity.id = position
            workEntity.title = "Work $position"
            workEntity.description = makeDetails(position)
            return workEntity
        }

        private fun makeDetails(position: Int): String {

            val builder = StringBuilder()
            builder.append("Details about Work: ")
                    .append(position)
            for (i in 0 until position) {
                builder.append("\nMore contactDetails information here.")
            }
            return builder.toString()
        }
    }
}
